homeassistant/components/teslemetry/sensor.py

```diff
@@ -0,0 +1,446 @@
+"""Sensor platform for Teslemetry integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    EntityCategory,
+    UnitOfElectricCurrent,
+    UnitOfElectricPotential,
+    UnitOfEnergy,
+    UnitOfLength,
+    UnitOfPower,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .entity import (
+    TeslemetryEnergyEntity,
+    TeslemetryVehicleEntity,
+    TeslemetryWallConnectorEntity,
+)
+from .models import TeslemetryEnergyData, TeslemetryVehicleData
+
+
+@callback
+def minutes_to_datetime(value: StateType) -> datetime | None:
+    """Convert relative minutes into absolute datetime."""
+    if isinstance(value, (int, float)) and value > 0:
+        return dt_util.now() + timedelta(minutes=value)
+    return None
+
+
+@dataclass(frozen=True, kw_only=True)
+class TeslemetrySensorEntityDescription(SensorEntityDescription):
+    """Describes Teslemetry Sensor entity."""
+
+    value_fn: Callable[[StateType], StateType | datetime] = lambda x: x
+
+
+VEHICLE_DESCRIPTIONS: tuple[TeslemetrySensorEntityDescription, ...] = (
```

Comment:
Consider to make some disabled by default? That's a lot of sensors...

Reply:
This isn't even the full list of sensors I'm going to be adding.

I'll see which ones are logical to disable.

Reply:
I think here with so many it might be good to be a bit conservative and make a good note on the doc instead that many sensors comes disabled or something to encourage the user to look at which makes sense to them.

Reply:
I have disabled a bunch now, I'll update the documentation accordingly.